compiler/test/tests/cobalt/Parser/MethodParserTest.hs

```diff
@@ -13,24 +13,69 @@ testMethodParser :: Test
+
+    let codeModifierProtected = "protected let exampleMethod (a: Int, b: Int): Int = _"
+    let testModifierProtected = testParseSuccess codeModifierProtected (MethodDef (Method {methodName = Name "exampleMethod", methodAnns = [], methodParams = [Field {fieldName = Name "a", fieldType = Just (TypeRef (RefLocal (Name "Int"))), fieldInit = Nothing},Field {fieldName = Name "b", fieldType = Just (TypeRef (RefLocal (Name "Int"))), fieldInit = Nothing}], methodModifiers = [Protected], methodReturnType = TypeRef (RefLocal (Name "Int")), methodBody = ExprAssignment (Identifier (Name "_"))})) methodDefParser
+    let testModifierProtectedStmt = testParseSuccess codeModifierProtected (MethodDef (Method {methodName = Name "exampleMethod", methodAnns = [], methodParams = [Field {fieldName = Name "a", fieldType = Just (TypeRef (RefLocal (Name "Int"))), fieldInit = Nothing},Field {fieldName = Name "b", fieldType = Just (TypeRef (RefLocal (Name "Int"))), fieldInit = Nothing}], methodModifiers = [Protected], methodReturnType = TypeRef (RefLocal (Name "Int")), methodBody = ExprAssignment (Identifier (Name "_"))})) statementParser
+
+    let codeModifierLocal = "local let exampleMethod (a: Int, b: Int): Int = _"
+    let testModifierLocal = testParseSuccess codeModifierLocal (MethodDef (Method {methodName = Name "exampleMethod", methodAnns = [], methodParams = [Field {fieldName = Name "a", fieldType = Just (TypeRef (RefLocal (Name "Int"))), fieldInit = Nothing},Field {fieldName = Name "b", fieldType = Just (TypeRef (RefLocal (Name "Int"))), fieldInit = Nothing}], methodModifiers = [PackageLocal], methodReturnType = TypeRef (RefLocal (Name "Int")), methodBody = ExprAssignment (Identifier (Name "_"))})) methodDefParser
+    let testModifierLocalStmt = testParseSuccess codeModifierLocal (MethodDef (Method {methodName = Name "exampleMethod", methodAnns = [], methodParams = [Field {fieldName = Name "a", fieldType = Just (TypeRef (RefLocal (Name "Int"))), fieldInit = Nothing},Field {fieldName = Name "b", fieldType = Just (TypeRef (RefLocal (Name "Int"))), fieldInit = Nothing}], methodModifiers = [PackageLocal], methodReturnType = TypeRef (RefLocal (Name "Int")), methodBody = ExprAssignment (Identifier (Name "_"))})) statementParser
+
+    let codeModifierMultiple = "final abstract public protected local let exampleMethod (a: Int, b: Int): Int = _"
+    let testModifierMultiple = testParseSuccess codeModifierMultiple (MethodDef (Method {methodName = Name "exampleMethod", methodAnns = [], methodParams = [Field {fieldName = Name "a", fieldType = Just (TypeRef (RefLocal (Name "Int"))), fieldInit = Nothing},Field {fieldName = Name "b", fieldType = Just (TypeRef (RefLocal (Name "Int"))), fieldInit = Nothing}], methodModifiers = [Final, Abstract, Public, Protected, PackageLocal], methodReturnType = TypeRef (RefLocal (Name "Int")), methodBody = ExprAssignment (Identifier (Name "_"))})) methodDefParser
```

Comment:
Is this supposed to be parsed this way? What are the semantics of using multiple contradicting modifiers?

Reply:
Currently the modifier parser looks like this.
```
modifiersParser :: Parser [Modifier]
modifiersParser = many $ choice [accessModifierParser, abstractModifierParser, finalModifierParser]
```
The issue is that the modifiers can be in any order and some can contain multiple. This means with parsing it will parse all of them even if there are too many (as you pointed out with `public`, `protected` and `final`.    
Unless there is a better way of doing this it would have to be checked when performing error checks on the AST.

Reply:
It seems like without forcing some arbitrary order of modifiers there is no way out of this. It will definitely need to be checked during semantic analysis.